csrc/multidevice/utils.h

```diff
@@ -86,7 +86,30 @@ bool haveDifferentShardings(
 // Returns whether a resharding expr reshards an inner axis
 bool isInnerResharding(Expr* expr);
 
-void shardAllLike(TensorView* ref, std::vector<TensorView*> tvs);
+// Shards all tensors in tvs like reference
+void shardAllLike(
+    TensorView* ref,
+    std::vector<TensorView*> tvs,
+    const std::unordered_set<ParallelType>& parallel_types = {
+        ParallelType::Serial,
+        ParallelType::DIDx});
+
+// Shards all TVs from the output of from expressions to the outputs of the to
+// expressions like the reference TV.
+void shardBetween(
+    const std::vector<Expr*>& from,
+    const std::vector<Expr*>& to,
+    TensorView* ref,
+    const std::unordered_set<ParallelType>& parallel_types = {
```

Comment:
Do you plan to customize this? It's always DIDx currently, and I'd expect this to be all DIDs for the future.

Reply:
For now it's just DIDx, but later we might want one TV to be the ref for DIDx and another for DIDy. 
Since it's not necessary now, I'm open to removing it and revisiting later.

Reply:
Got it. I'd go YAGNI if (I think so) it leads to less code.